apps/vaporgui/PCaptureWidget.cpp

```diff
@@ -16,119 +17,124 @@
 #include <QFileDialog>
 #include <QHBoxLayout>
 
-const std::string CaptureModes::CURRENT = "Current frame";
-const std::string CaptureModes::RANGE   = "Timeseries range";
-
-const std::string TiffStrings::CaptureFileType = "TIFF";
 const std::string TiffStrings::FileFilter = "TIFF (*.tif)";
 const std::string TiffStrings::FileSuffix = "tif";
 
-const std::string PngStrings::CaptureFileType = "PNG";
 const std::string PngStrings::FileFilter = "PNG (*.png)";
 const std::string PngStrings::FileSuffix = "png";
 
-PCaptureToolbar::PCaptureToolbar(VAPoR::ControlExec *ce, MainForm *mf)
+PCaptureHBox::PCaptureHBox(VAPoR::ControlExec *ce, MainForm *mf)
     : PWidget("", _hBox = new VHBoxWidget()), _ce(ce), _mf(mf)
 {
     _hBox->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Preferred);
 
-    _typeCombo = new VComboBox({TiffStrings::CaptureFileType, PngStrings::FileFilter});
+    _typeCombo = new PEnumDropdownStandalone(AnimationParams::CaptureTypeTag, {"TIFF", "PNG"}, {0, 1});
+
     _fileLabel = new VLabel("");
+    _fileLabel->MakeSelectable();
 
     _captureButton = new PButton("Capture\nCurrent Frame", [this](VAPoR::ParamsBase*){_captureSingleImage();});
-    // PWidget is setting _enableBasedOnParamsValue to 0, so we need to set it to a non-zero value.
-    // The default value is not being set and should be garbage.  Why/how is it being initialized to 0?
-    _captureButton->ShowBasedOnParam(AnimationParams::CaptureModeTag, -1);
-    _captureButton->ShowBasedOnParam(AnimationParams::CaptureModeTag, CaptureModes::CURRENT);
+    _captureButton->ShowBasedOnParam(AnimationParams::CaptureModeTag, 0);
 
-    _captureTimeseriesButton = new PButton("Capture\nTimeseries", [this](VAPoR::ParamsBase*){_captureTimeseries();});
-    // Need to set ShowBasedOnParam for a numeric value.  See previous comment.
-    _captureTimeseriesButton->ShowBasedOnParam(AnimationParams::CaptureModeTag, -1);
-    _captureTimeseriesButton->ShowBasedOnParam(AnimationParams::CaptureModeTag, CaptureModes::RANGE);
-    
+    _captureTimeSeriesButton = new PButton("Capture\nTime Series", [this](VAPoR::ParamsBase*){_captureTimeSeries();});
+    _captureTimeSeriesButton->ShowBasedOnParam(AnimationParams::CaptureModeTag, 1);
 
     QHBoxLayout* layout = qobject_cast<QHBoxLayout*>(_hBox->layout());
     layout->addWidget(_typeCombo,1);
     layout->addWidget(_fileLabel,3);
     layout->addWidget(_captureButton,1);
-    layout->addWidget(_captureTimeseriesButton,1);
+    layout->addWidget(_captureTimeSeriesButton,1);
+}
+
+void PCaptureHBox::_dropdownIndexChanged(int index) {
+    int value;
+    if (_enumMap.empty()) {
+        value = index;
+    } else {
+        VAssert(index >= 0 && index < _enumMap.size());
+        value = _enumMap[index];
```

Comment:
The VAssert is intended to be used in scenarios where something should crash in debug but silently default or alert in release builds but this will crash in both cases.